### PR TITLE
[Serve][Doc] Split core-apis to key concepts and user guide

### LINKED ISSUE #22872: [docs] [serve] Create "key concepts" page
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

This is where "Serve Architecture" and the concepts currently in "Core API: Deployments" should go. The latter should probably be converted into a user guide called "Using Deployments". The word "core" is also extremely confusing.

### PR DESCRIPTION
**Pull request by simon-mo**
## Why are these changes needed?
This PR splits the old "core-apis" page which is _just_ about deployments into a key concept page and two user guides. 

Additionally, the following refactoring:
> Key concepts: {Deployment, HTTP Ingress, ServeHandle, Deployment Graph}
User guides: {Managing Deployments, Dependencies, HTTP, Composing Deployments, Serving ML Models, Deploying Ray Serve, Debugging/Monitoring, Deployment Graph, Performance Tuning}
"Advanced Tutorials" -> "Examples"

## Related issue number
Closes #22872

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
Would also suggest we add deployment graph to this page while we're at it.

**Comment by edoakes** [comment]
Synced offline. End state for this PR:

- Key concepts: {Deployment, HTTP Ingress, ServeHandle, Deployment Graph}
- User guides: {Managing Deployments, Dependencies, HTTP, Composing Deployments, Serving ML Models, Deploying Ray Serve, Debugging/Monitoring, Deployment Graph, Performance Tuning}
- "Advanced Tutorials" -> "Examples"

Let's not worry about modifying content for now.

**Comment by edoakes** [comment]
@simon-mo waiting for docs build to work, can you remove `@author-action-required` when it's fixed?

**Comment by simon-mo** [comment]
@edoakes doc built

**Comment by edoakes** [comment]
@simon-mo nice, this is looking good! Can you please link out to the relevant user guides for each of the "key concepts?"

**Comment by simon-mo** [comment]
I modified it to put the a "what's next" section at the end of key concept doc instead. because user need to understand at least two key concept first before diving deep.

File: doc/source/serve/key-concepts.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+```{literalinclude} doc_code/key-concepts-deployment-graph.py
+```
+
+## What's Next?
+Now you have learned about the key concept. You can dive into our [User Guides](user-guide) for more details into:
+- [Create/Update/Delete Serve Deployments](managing-deployments)
+- [Configuring HTTP Ingress and Bring FastAPI App](http-guide)
+- [Composing Deployments with ServeHandle](handle-guide)
+- [Building Deployment Graph](deployment-graph)
```
Comment (edoakes) [inline doc/source/serve/key-concepts.md:121]:
```suggestion
- [Creating, updating, and deleting deployments](managing-deployments)
- [Configuring HTTP ingress and integrating with FastAPI](http-guide)
- [Composing deployments using ServeHandle](handle-guide)
- [Building Deployment Graphs](deployment-graph)
```

**Review by edoakes — APPROVED** [review]

**Comment by pcmoritz** [comment]
This broke linkcheck I think, can you fix it @simon-mo ?

**Comment by simon-mo** [comment]
Thanks for catching. Will do!

**Comment by simon-mo** [comment]
Also please feel free to revert.

**Comment by pcmoritz** [comment]
linkcheck failures don't need to be reverted, but we should fix them in a timely manner as we detect them

**Comment by simon-mo** [comment]
hmmm it auto-magically recovered

image

**Pull request merged**